tests/Unit/SearchEngine/Driver/Elasticsearch/IndexUpdaterTest.php

```diff
@@ -0,0 +1,395 @@
+    {
+        $this->client->expects($this->once())
+            ->method('updateByQuery')
+            ->with(
+                [
+                    'index' => 'items',
+                    'type' => '_doc',
+                    'conflicts' => 'proceed',
+                    'wait_for_completion' => true,
+                    'body' => [
+                        'query' => [
+                            'multi_match' => [
+                                'query' => TaoOntology::CLASS_URI_ITEM,
+                                'fields' => ['type', '_id'],
+                            ],
+                        ],
+                        'script' => [
+                            'source' => $source
+                        ]
+                    ]
+                ]
+            );
+        $this->sut->deleteProperty($property);
+    }
+
+    public function testExceptionWhenRemovingProperty(): void
+    {
+        $this->expectException(FailToRemovePropertyException::class);
+
+        $this->client->expects($this->once())
+            ->method('updateByQuery')
+            ->willThrowException(new BadMethodCallException());
+
+        $this->sut->deleteProperty(
+            [
+                'name' => 'test',
+                'parentClasses' => ['someClass'],
+                'type' => TaoOntology::CLASS_URI_ITEM
+            ]
+        );
+    }
+
+    /**
+     * @dataProvider provideInvalidPropertiesForRemoval
+     */
+    public function testDoNotRemoveProperty(array $property): void
+    {
+        $this->client->expects($this->never())
+            ->method('updateByQuery');
+
+        $this->sut->deleteProperty($property);
+    }
+
+    public function testUpdatePropertyValueSuccessfuly()
+    {
+        $documentUri = 'https://tao.docker.localhost/ontologies/tao.rdf#i5ef45f413088c8e7901a84708e84ec';
+        $validType = 'http://www.tao.lu/Ontologies/TAOItem.rdf#Item';
+        $source = 'ctx._source[\'RadioBox_isTest\'] = [\'123456\'];';
+
+        $this->client->expects($this->once())
+            ->method('updateByQuery')
+            ->with(
+                [
+                    'index' => 'items',
+                    'type' => '_doc',
+                    'conflicts' => 'proceed',
+                    'wait_for_completion' => true,
+                    'body' => [
+                        'query' => [
+                            'multi_match' => [
+                                'query' => $documentUri,
+                                'fields' => ['type', '_id'],
+                            ],
+                        ],
+                        'script' => [
+                            'source' => $source
+                        ]
+                    ]
+                ]
+            );
+
+        $this->sut->updatePropertyValue(
+            $documentUri,
+            [
+                $validType
+            ],
+            'RadioBox_isTest',
+            ['123456']
+        );
```

Comment:
It is better to either declare all arrays in this method call in one line or spit them in multiple lines.

```suggestion
        $this->sut->updatePropertyValue(
            $documentUri,
            [$validType],
            'RadioBox_isTest',
            ['123456']
        );
```